plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/file/FileMetastoreTableOperations.java

```diff
@@ -76,7 +80,13 @@ protected void commitToExistingTable(TableMetadata base, TableMetadata metadata)
             metastore.replaceTable(database, tableName, table, privileges);
         }
         catch (RuntimeException e) {
-            throw new CommitFailedException(e, "Failed to commit transaction to FileHiveMetastore");
+            if (e instanceof TrinoException trinoException &&
+                    trinoException.getErrorCode() == HIVE_METASTORE_ERROR.toErrorCode() &&
+                    Objects.equals(trinoException.getMessage(), "Cannot update Iceberg table: supplied previous location does not match current location")) {
+                // CommitFailedException is handled as a special case in the Iceberg library. This commit will automatically retry
```

Comment:
nit: the essence of this comment for me is that this is the only scenario in which it's safe to delete the metadata files from the FS and retry (in all other scenarios it's not possible to tell with certainty whether the commit swap succeeded, so it's up to the user to ascertain)

Reply:
@marton-bod 

> the essence of this comment for me is that this is the only scenario in which it's safe to delete the metadata files from the FS and retry

The metadata files are removed only after exhausting the retries.

Note that [cleanUpOnCommitFailure ](https://github.com/apache/iceberg/blob/cc7e6d9fa259707a5c690b3534ea6b5c7da3fc05/core/src/main/java/org/apache/iceberg/BaseTransaction.java#L402) is thrown only after exhausting the retries.